#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "merged_depth_cam"

gen = ParameterGenerator()

gen.add("target_height", int_t, 0, "The size of the depth blobs drawn on the resulting image", 640, 0, 10000)
gen.add("target_width", int_t, 0, "The size of the depth blobs drawn on the resulting image", 480, 0, 10000)
gen.add("cx", double_t, 0, "Principal point cx", 0.0, -10000.0, 10000.0)
gen.add("cy", double_t, 0, "Principal point cy", 0.0, -10000.0, 10000.0)
gen.add("fx", double_t, 0, "The focal length along x of the virtual depth cam", 570.0, 0.001, 10000.0)
gen.add("fy", double_t, 0, "The focal length along y of the virtual depth cam", 570.0, 0.001, 10000.0)

exit(gen.generate(PACKAGE, "merged_depth_cam", "ModCamInfo"))
